components/collector/src/base_collectors/collector.py

```diff
@@ -108,3 +101,8 @@ def __should_collect(self, metric_uuid: str, metric: dict) -> bool:
         metric_edited = self.__previous_metrics.get(metric_uuid) != metric
         metric_due = self.next_fetch.get(metric_uuid, datetime.min.replace(tzinfo=UTC)) <= datetime.now(tz=UTC)
         return metric_edited or metric_due
+
+    def __log_tasks(self, event: str, warning_threshold: int = 0) -> None:
+        """Log the number of running tasks."""
+        level = logging.WARNING if len(self.tasks) > warning_threshold else logging.INFO
+        logging.log(level, "Collector %s: %d task(s) currently running", event, len(self.tasks))
```

Comment:
Can we perhaps stick this in a separate logger, so it doesn't clutter the regular logs?

Reply:
We could, but that seems outside the scope of this issue.